featuretools/tests/primitive_tests/test_primitive_utils.py

```diff
@@ -135,3 +138,94 @@ def test_errors_no_primitive_in_file(bad_primitives_files_dir):
     with pytest.raises(RuntimeError) as excinfo:
         load_primitive_from_file(primitive_file)
     assert str(excinfo.value) == error_text
+
+
+@pytest.mark.parametrize(
+    "window_length, gap",
+    [
+        (3, 2),
+        (3, 4),  # gap larger than window
+        (2, 0),  # gap explicitly set to 0
+    ],
+)
+def test_roll_series_with_gap(window_length, gap, rolling_series_pd):
+    rolling_max = to_pandas(_roll_series_with_gap(rolling_series_pd, window_length, gap=gap).max())
+    rolling_min = to_pandas(_roll_series_with_gap(rolling_series_pd, window_length, gap=gap).min())
+
+    assert len(rolling_max) == len(rolling_series_pd)
+    assert len(rolling_min) == len(rolling_series_pd)
+
+    for i in range(len(rolling_series_pd)):
+        start_idx = i - gap - window_length + 1
+        end_idx = i - gap
+
+        # If start and end are negative, they're entirely before
+        if start_idx < 0 and end_idx < 0:
+            assert pd.isnull(rolling_max.iloc[i])
+            assert pd.isnull(rolling_min.iloc[i])
+            continue
+
+        if start_idx < 0:
+            start_idx = 0
+
+        # Because the row values are a range from 0 to 20, the rolling min will be the start index
+        # and the rolling max will be the end idx
+        assert rolling_min.iloc[i] == start_idx
+        assert rolling_max.iloc[i] == end_idx
+
+
+def test_roll_series_with_no_gap(rolling_series_pd):
+    window_length = 3
+    gap = 0
+    actual_rolling = _roll_series_with_gap(rolling_series_pd, window_length, gap=gap).mean()
+    expected_rolling = rolling_series_pd.rolling(window_length, min_periods=1).mean()
+
+    pd.testing.assert_series_equal(to_pandas(actual_rolling), to_pandas(expected_rolling))
+
+
+@pytest.mark.parametrize(
+    "window_length, gap",
+    [
+        (6, 2),
+        (6, 0)  # No gap - changes early values
+    ]
+)
+def test_roll_series_with_gap_early_values(window_length, gap, rolling_series_pd):
+    # Default min periods is 1 - will include all
+    default_partial_values = to_pandas(_roll_series_with_gap(rolling_series_pd,
```

Comment:
Leftover `to_pandas` call?

Reply:
yep, will remove all the to_pandas calls in this file!